docs/creating_integration_manifest.md

```diff
@@ -92,7 +92,7 @@ If this integration is being submitted for inclusion in Home Assistant, it shoul
 
 ## Dependencies
 
-Dependencies are other Home Assistant integrations that you want Home Assistant to set up successfully prior to the integration being loaded. This can be necessary in case you want to offer functionality from that other integration, like using webhooks or an MQTT connection.
+Dependencies are other Home Assistant integrations you want Home Assistant to set up successfully before the integration is loaded. Adding an integration to dependencies will ensure the depending integration is loaded before setup, but it does not guarantee all dependency configuration entries have been set up. Adding dependencies can be necessary if you want to offer functionality from that other integration, like webhooks or an MQTT connection. Adding an [after dependency](#after-dependencies) might be a better alternative if a dependency is optional but not critical. See the [MQTT section](#mqtt) for more details on handling this for MQTT.
```

Comment:
Related code for anyone looking at this 


https://github.com/home-assistant/core/blob/41515249a0ba3a4a24e3fe55dd2c9a9dc46a798f/homeassistant/setup.py#L295

https://github.com/home-assistant/core/blob/95023ecf08f01b5ebb88fb0dab285263dd435439/homeassistant/setup.py#L125